# commit 76a66d510a3737674563133a420f4fd22da42c1b
# Author: Anton Blanchard <anton@au1.ibm.com>
# Date:   Sat Aug 17 18:33:02 2013 +0930
# 
#     PowerPC floating point little-endian [14 of 15]
#     http://sourceware.org/ml/libc-alpha/2013-07/msg00205.html
#     
#     These all wrongly specified float constants in a 64-bit word.
#     
#         * sysdeps/powerpc/powerpc64/fpu/s_ceilf.S: Correct float constants
#         for little-endian.
#         * sysdeps/powerpc/powerpc64/fpu/s_floorf.S: Likewise.
#         * sysdeps/powerpc/powerpc64/fpu/s_nearbyintf.S: Likewise.
#         * sysdeps/powerpc/powerpc64/fpu/s_rintf.S: Likewise.
#         * sysdeps/powerpc/powerpc64/fpu/s_roundf.S: Likewise.
#         * sysdeps/powerpc/powerpc64/fpu/s_truncf.S: Likewise.
#
---
# sysdeps/powerpc/powerpc64/fpu/s_ceilf.S      |    4 +++-
# sysdeps/powerpc/powerpc64/fpu/s_floorf.S     |    4 +++-
# sysdeps/powerpc/powerpc64/fpu/s_nearbyintf.S |    4 +++-
# sysdeps/powerpc/powerpc64/fpu/s_rintf.S      |    4 +++-
# sysdeps/powerpc/powerpc64/fpu/s_roundf.S     |    6 ++++--
# sysdeps/powerpc/powerpc64/fpu/s_truncf.S     |    4 +++-
# 6 files changed, 19 insertions(+), 7 deletions(-)
#
--- a/sysdeps/powerpc/powerpc64/fpu/s_ceilf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_ceilf.S
@@ -19,8 +19,10 @@
 #include <sysdep.h>
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
+	.long 0x0
 	.section	".text"
 
 EALIGN (__ceilf, 4, 0)
--- a/sysdeps/powerpc/powerpc64/fpu/s_floorf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_floorf.S
@@ -19,8 +19,10 @@
 #include <sysdep.h>
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
+	.long 0x0
 	.section	".text"
 
 EALIGN (__floorf, 4, 0)
--- a/sysdeps/powerpc/powerpc64/fpu/s_nearbyintf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_nearbyintf.S
@@ -26,8 +26,10 @@
 /* float [fp1] nearbyintf(float [fp1]) */
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
+	.long 0x0
 	.section	".text"
 
 EALIGN (__nearbyintf, 4, 0)
--- a/sysdeps/powerpc/powerpc64/fpu/s_rintf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_rintf.S
@@ -19,8 +19,10 @@
 #include <sysdep.h>
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
+	.long 0x0
 	.section	".text"
 
 EALIGN (__rintf, 4, 0)
--- a/sysdeps/powerpc/powerpc64/fpu/s_roundf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_roundf.S
@@ -19,10 +19,12 @@
 #include <sysdep.h>
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
 .LC1:	/* 0.5 */
-	.tc FD_3f000000_0[TC],0x3f00000000000000
+	.long 0x3f000000
+
 	.section	".text"
 	
 /* float [fp1] roundf  (float x [fp1])
--- a/sysdeps/powerpc/powerpc64/fpu/s_truncf.S
+++ b/sysdeps/powerpc/powerpc64/fpu/s_truncf.S
@@ -19,8 +19,10 @@
 #include <sysdep.h>
 
 	.section	".toc","aw"
+	.p2align 3
 .LC0:	/* 2**23 */
-	.tc FD_4b000000_0[TC],0x4b00000000000000
+	.long 0x4b000000
+	.long 0x0
 	.section	".text"
 	
 /* float [fp1] truncf (float x [fp1])
